src/components/solvers/SolveRequest.ts

```diff
@@ -0,0 +1,9 @@
+export interface SolveRequest {
+    requestContent?: any;
```

Comment:
Content can be optional because it's used in nested `SolverPicker` components which are used to pick subroutine solvers. Those don't have any content. Technically, only the first SolveRequest has a content.

To solve this, I can create an interface `SubSolveRequest` without `requestContent` which `SolveRequest` extends, and use it in the `SolverPicker`.

Reply:
That's better! But I think `requestContent` in `SolveRequest` can be non-optional now (line 13).

I think the `extends` relationship between `SubSolveRequest` (superclass) and `SolveRequest` (subclass) is a bit counterintuitive at the moment although it is technically correct. Maybe you can rename `SubSolveRequest` to `SolverChoice`? ("SolveRequest is-a SolverChoice" seems to make more sense to me).